### PR TITLE
Adding the documentation for assert_no_cpu_op #2881.

### PR DESCRIPTION
**Pull request by caglar**
Started writing the documentation for the assert_no_cpu_op.

### CODE REVIEW

File: doc/tutorial/debug_faq.txt [hunk 1]
```diff
@@ -295,6 +295,23 @@ Tips:
...
+is a CPU Op on your computational graph. assert_no_cpu_op can take the following one of the three 
+options:
+
+* ``warn``: Raise a warning
+* ``pdb``: Stop with a pdb in the computational graph during the compilation
+* ``raise``: Raise an error,
+if there is a CPU Op in the computational graph.
+
+It is possible to use this mode by providing the flag in THEANO_FLAGS, such as:
+``THEANO_FLAGS="float32,assert_no_cpu_op='raise'" python test.py``
```
Comment (nouiz) [inline doc/tutorial/debug_faq.txt]: add device=gpu. Tell that we don't always print them, and I think we still miss some detection pattern.

File: doc/tutorial/debug_faq.txt [hunk 2]
```diff
@@ -295,6 +295,23 @@ Tips:
   of type *float64*.
 
 
+"Why is my GPU function seems to be slow?"
+------------------------------------
```
Comment (lamblin) [inline doc/tutorial/debug_faq.txt]: I would say "Why does my GPU function seem to be slow?"

Comment (lamblin) [inline doc/tutorial/debug_faq.txt]: Also, you may need to add more "-" to be as wide as the title.

**Comment by lamblin** [comment]
Otherwise, once you address Fred's comments, I think it is good. Thanks!

**Comment by caglar** [comment]
OK I have fixed the parts you commented.

**Comment by nouiz** [comment]
I killed travis as it isn't useful.

**Comment by lamblin** [comment]
Thanks!

**Pull request merged**